aten/src/ATen/native/Convolution.cpp

```diff
@@ -147,11 +147,12 @@ auto ConvParams::use_miopen(const at::Tensor& input) const -> bool {
 
 auto ConvParams::use_mkldnn(const at::Tensor& input) const -> bool {
 #if AT_MKLDNN_ENABLED()
-  return input.type().backend() == at::Backend::CPU &&
-         input.scalar_type() == kFloat && // only on CPU Float Tensors
-         !is_dilated() && // doesn't support dilation
-         !transposed && // or transposed tensors
-         input.ndimension() == 4; // must be in NCHW format
+  return (input.type_id() == MkldnnCPUTensorId()) || // input is mkldnn Tensor
+    (input.type().backend() == at::Backend::CPU &&
+     input.scalar_type() == kFloat && // only on CPU Float Tensors
+     !is_dilated() && // doesn't support dilation
+     !transposed && // or transposed tensors
+     input.ndimension() == 4); // must be in NCHW format
```

Comment:
For the comment, the original 'NCHW format' does not stand now.
BTW, conv with dilation and ndim=3/4/5 (aka. 1D, 2D, 3D) all supported by mkldnn. But we can do it with a separate PR.

Reply:
These checks are for the conv on CPU tensor. Probably we just keep what has been done on the original path and put all future opts in the new MKL-DNN tensor path?

Reply:
Oh, that is for CPU tensor. Then forget what I said :-)

Reply:
what's the story with the other checks?  We can't use CPU tensor / MKLDNN algo if it's dilated / transposed / or the input dimension is wrong, but we can if it's an MKLDNN tensor?

And where's the check if it's not conv2d -- what happens if you try to call one of the other algos with an MKLDNN tensor?